roles/pre_tasks/tasks/main.yml

```diff
@@ -1,10 +1,23 @@
 ---
 
 # Install absolutely necessary packages
+
+- name: Check if the absolutely necessary packages are installed
+  shell: "type {{ item }}"
+  with_items: "{{ tower_pre_tasks_packages }}"
+  register: type_packages
+  ignore_errors: yes
+
 - name: "Install absolutely necessary packages"
   yum:
-    name: "{{ tower_pre_tasks_packages }}"
-    state: present
+    name: "{{ package_item.item }}"
+    state: latest
+  ignore_errors: true
```

Comment:
you shouldn't need to ignore errors here? You want to fail if a required package can't be installed.

Reply:
I'm trying to remember why I did this, I think because it failed when you don't have root access, Will do testing to see about removing

Reply:
Updated to remove, the when should handle this case